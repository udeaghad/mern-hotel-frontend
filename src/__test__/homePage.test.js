import React from 'react';
import { Provider } from  'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/ConfigureStore'
import Homepage from '../components/HomePage'
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';


describe("To test the home page component", () =>{
  test('To test Home page renders successfully!', () => {    
    const tree = renderer
      .create(
      <BrowserRouter>
        <Provider store={store}>
        <Homepage />
      </Provider>
      </BrowserRouter>
      )
      .toJSON;
    expect(tree).toMatchSnapshot()
  })

  test('Should display a welcome text', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <Provider store={store}>
          <Homepage />
        </Provider>
      </BrowserRouter>
      )
    const text = getByTestId('heading-text')
    
    expect(text.innerHTML).toBe("...you wanna book a hotel")
  })

  test('Should display the carousel', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <Provider store={store}>
          <Homepage />
        </Provider>
      </BrowserRouter>
      )
    const text = getByTestId('main_card_container')    
    expect(text.innerHTML).toContain("carousel")
  })
})
