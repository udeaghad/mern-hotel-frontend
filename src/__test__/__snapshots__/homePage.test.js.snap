// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the home page component To test Home page renders successfully! 1`] = `[Function]`;

exports[`To test thehome page component To test Home page renders successfully! 1`] = `[Function]`;
